fhir-server-test/src/test/java/com/ibm/fhir/server/test/TenantIsolationTest.java

```diff
@@ -0,0 +1,116 @@
+import com.ibm.fhir.model.test.TestUtil;
+import com.ibm.fhir.model.type.Code;
+import com.ibm.fhir.model.type.Extension;
+import com.ibm.fhir.model.type.Narrative;
+import com.ibm.fhir.model.type.Uri;
+import com.ibm.fhir.model.type.code.CodeSystemContentMode;
+import com.ibm.fhir.model.type.code.ExtensionContextType;
+import com.ibm.fhir.model.type.code.IssueSeverity;
+import com.ibm.fhir.profile.ExtensionBuilder;
+
+public class TenantIsolationTest extends FHIRServerTestBase {
+    @BeforeClass
+    public void createArtifactsInDefaultTenant() throws Exception {
+        Context context = Context.builder()
+                .type(ExtensionContextType.ELEMENT)
+                .expression("Element")
+                .build();
+        StructureDefinition extension = new ExtensionBuilder("http://ibm.com/fhir/StructureDefinition/favorite-team", "0.0.1", context, "string")
+                .build();
+        createResourceAndReturnTheLogicalId("StructureDefinition", extension);
+
+        CodeSystem codeSystem = TestUtil.getMinimalResource(CodeSystem.class).toBuilder()
+                .url(Uri.of("http://ibm.com/fhir/StructureDefinition/teams"))
+                .version("0.0.1")
+                .content(CodeSystemContentMode.COMPLETE)
+                .concept(Concept.builder()
+                    .code(Code.of("Packers"))
+                    .build())
+                .concept(Concept.builder()
+                    .code(Code.of("Brewers"))
+                    .build())
+                .concept(Concept.builder()
+                    .code(Code.of("Bucks"))
+                    .build())
+                .concept(Concept.builder()
+                    .code(Code.of("Badgers"))
+                    .build())
+                .build();
+        createResourceAndReturnTheLogicalId("CodeSystem", codeSystem);
+    }
+
+    @Test()
+    public void testCodeSystemIsolation() throws Exception {
+        Parameters params = Parameters.builder()
+            .parameter(Parameter.builder()
+                .name("system")
+                .value(Uri.of("http://ibm.com/fhir/StructureDefinition/teams"))
+                .build())
+            .parameter(Parameter.builder()
+                .name("code")
+                .value(Code.of("Packers"))
+                .build())
+            .build();
+
+        FHIRResponse response = client.invoke("CodeSystem", "$lookup", params);
+        Resource resource = response.getResource(Resource.class);
+        assertTrue(resource instanceof Parameters);
+        System.out.println(resource);
+
+        response = client.invoke("CodeSystem", "$lookup", params,
+            FHIRRequestHeader.header("X-FHIR-TENANT-ID", "tenant1"),
+            FHIRRequestHeader.header("X-FHIR-DSID", "reference"));
+        OperationOutcome oo = response.getResource(OperationOutcome.class);
+        assertEquals(oo.getIssue().get(0).getSeverity(), IssueSeverity.FATAL, oo.getIssue().get(0).getDetails().toString());
```

Comment:
should there be a positive case? e.g. one that succeeds to prove the structuredefinition/extension above was correct.

Reply:
I was also thinking we should push two structuredefinitions with the same name and two distinct behaviors.

Reply:
this is what lines 80 through 83 are doing

Reply:
Right, so the same named structure def would be the only stretch goal I could for see

Reply:
though I should probably remove that sysout...
I can throw a `assertEquals(response.getStatus(), 200);` in there while at it for good measure

Reply:
ok, I restructured the test completely.  please have another look; i'll plan to merge if everything passes

Reply:
There was a challenge in that the ServerRegistryResourceProvider caches results for a minute.
Fortunately, that cache is kept on a per-datastore basis and so we can avoid the trouble by using different tenant1 datastores for the two ("no definition" vs "conflicting definition") cases.

Reply:
nice